### ISSUE TITLE
Custom local variable display hook

### ISSUE THREAD

**Issue by leifwalsh** [twosigma/marbles#96]
**Is your feature request related to a problem? Please describe.**

See https://github.com/twosigma/marbles/issues/92#issuecomment-423780916

We should provide a hook users can implement on their classes to customize display of values in failure messages.

**Describe the solution you'd like**
If a class implements `_repr_marbles_`, call that method instead of `repr` to create the string to display in failure messages.

**Comment by leifwalsh** [comment]
@vegarsti I filled out the description a little more, let me know if it makes sense and if you want to take over design more and flesh out the details.

**Comment by vegarsti** [comment]
Great!

**Comment by vegarsti** [comment]
Sorry for the silence here, been finishing studies. Will look at it soon!

**Comment by lvllvl** [comment]
Hi just wondering if this issue is still up for grabs? I don't see any PRs but I know someone mentioned they were working on this so I thought I'd check.

**Comment by leifwalsh** [comment]
Yep, this is up for grabs! We actually owe some release engineering work here too, I'd like to take a fresh look at how we build, test, and package this, and maybe add devcontainer support, but I'm not sure when I'll have the time. 

If you want to work on this and you get stuck, comment here or email me and I'll try to help out.
